code/CMSBatchAction.php

```diff
@@ -107,9 +108,14 @@ public function batchaction(SS_List $objs, $helperMethod, $successMessage, $argu
 
             // Now make sure the tree title is appropriately updated
             $publishedRecord = DataObject::get_by_id($this->managedClass, $id);
+            if ($publishedRecord instanceof SiteTree) {
+                $treeTitle = CMSMain::singleton()->getRecordTreeMarkup($publishedRecord);
```

Comment:
This looks like new hard coupling - but this class was always intended to be used with SiteTree.
The coupling will be removed in https://github.com/silverstripe/silverstripe-cms/issues/2950